recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc-caam.sh

```diff
@@ -61,6 +67,29 @@ tdx_enc_key_gen() {
     fi
 }
 
+# backup original data in partition (if not encrypted)
+tdx_enc_backup_data() {
+    if [ ${TDX_ENC_PRESERVE_DATA} -eq 1 ]; then
+        mkdir -p "${TDX_ENC_STORAGE_MOUNTPOINT}"
+        if mount ${TDX_ENC_STORAGE_LOCATION} "${TDX_ENC_STORAGE_MOUNTPOINT}"; then
+            tdx_enc_log "Backing up original content..."
+
+            TDX_ENC_PRESERVE_LOCATION=$(mktemp)
+            BACKUP_STORAGE_LIMIT="$(awk '/MemFree/ { printf "%lu", $2/30 }' /proc/meminfo)"
+            tdx_enc_log "Backup limit determined: ${BACKUP_STORAGE_LIMIT}"
+            
+            msgs="$(tar -C "${TDX_ENC_STORAGE_MOUNTPOINT}" -c . | bzip2 -cz | dd bs=1024 count=${BACKUP_STORAGE_LIMIT} of=${TDX_ENC_PRESERVE_LOCATION})"
```

Comment:
Not sure what to do here to ensure the errors are raised and captured properly.
See: https://github.com/toradex/meta-toradex-security/pull/27#discussion_r1606716879

Reply:
Not sure if @rborn-tx has any comments. I will have to think about this and do some tests. I will do this earlier next week.

Reply:
At this point I wanted to confirm: hasn't any of the options I mentioned along the way worked? As a recap, the options were:

1) To deal with the lack of pipefail in the shell:

```
msgs="$({ { tar -C "${TDX_ENC_STORAGE_MOUNTPOINT}" -cj . || echo "ERROR" >&2; } | dd bs=1024 count=100 of=/tmp/saved.tar.bz2; } 2>&1)"
if [ "$?" -ne 0 ] || echo "${msgs}" | grep -q ERROR; then
    # Some error happened.
fi
```

2) To deal with the above plus the apparent unavailable `-j` option in `tar`:

```
msgs="$({ { tar -C "${TDX_ENC_STORAGE_MOUNTPOINT}" -c . || echo "ERROR" >&2; } | { bzip2 -cz || echo "ERROR" >&2; } | dd bs=1024 count=100 of=/tmp/saved.tar.bz2; } 2>&1)"
if [ "$?" -ne 0 ] || echo "${msgs}" | grep -q ERROR; then
    # Some error happened.
fi
```

3) To deal with the above plus the apparent impossibility of capturing stderr into a shell variable:

```
{ { tar -C "${TDX_ENC_STORAGE_MOUNTPOINT}" -c . || echo "ERROR" >&2; } | { bzip2 -cz || echo "ERROR" >&2; } | dd bs=1024 count=100 of=/tmp/saved.tar.bz2; } 2>/tmp/errors.txt
if [ "$?" -ne 0 ] || cat /tmp/errors.txt | grep -q ERROR; then
    # Some error happened.
fi
```

If none of them worked, then I ran out of ideas and would need to debug the actual script.

Reply:
@rborn-tx - Every time I have added the `.. || echo "ERROR" >&2` I get errors when the script runs.
I also had to split apart the bzip from the tar command (as it never seems to work), however, that is already in the latest code in my branch. I can only assume this is another BusyBox config that I have not yet found to enable the single tar command to work.